packages/artifact/src/internal/download-http-client.ts

```diff
@@ -124,66 +160,124 @@ export class DownloadHttpClient {
         )
+        return
+      } else if (isRetryableStatusCode(response.message.statusCode)) {
+        info(
+          `A ${response.message.statusCode} response code has been received while attempting to download an artifact`
+        )
+        // if a throttled status code is received, try to get the retryAfter header value, else differ to standard exponential backoff
+        isThrottledStatusCode(response.message.statusCode)
+          ? await backOff(
+              tryGetRetryAfterValueTimeInMilliseconds(response.message.headers)
+            )
+          : await backOff()
       } else {
+        // Some unexpected response code, fail immediately and stop the download
         // eslint-disable-next-line no-console
-        console.log(retryResponse)
-        throw new Error(`Unable to download ${artifactLocation}`)
+        console.log(response)
+        throw new Error(
+          `###ERROR### Unable to download ${artifactLocation} ###`
+        )
       }
-    } else {
-      // eslint-disable-next-line no-console
-      console.log(response)
-      throw new Error(`Unable to download ${artifactLocation}`)
     }
   }
 
   /**
-   * Pipes the response from downloading an individual file to the appropriate stream
-   * @param response the http response recieved when downloading a file
-   * @param stream the stream where the file should be written to
-   * @param isGzip does the response need to be be uncompressed
+   * Pipes the response from downloading an individual file to the appropriate destination stream while decoding gzip content if necessary
+   * @param response the http response received when downloading a file
+   * @param destinationStream the stream where the file should be written to
+   * @param isGzip a boolean denoting if the content is compressed using gzip and if we need to decode it
    */
-  private async pipeResponseToStream(
+  private async pipeResponseToFile(
     response: IHttpClientResponse,
-    stream: NodeJS.WritableStream,
+    destinationStream: fs.WriteStream,
     isGzip: boolean
   ): Promise<void> {
-    return new Promise(resolve => {
+    await new Promise((resolve, reject) => {
       if (isGzip) {
-        // pipe the response into gunzip to decompress
         const gunzip = zlib.createGunzip()
         response.message
           .pipe(gunzip)
-          .pipe(stream)
+          .pipe(destinationStream)
           .on('close', () => {
             resolve()
           })
+          .on('error', error => {
+            info(
+              `An error has been encountered while gunzipping and writing a downloaded file to ${destinationStream.path}`
+            )
+            reject(error)
+          })
       } else {
-        response.message.pipe(stream).on('close', () => {
-          resolve()
-        })
+        response.message
+          .pipe(destinationStream)
+          .on('close', () => {
+            resolve()
+          })
+          .on('error', error => {
+            info(
```

Comment:
These errors should not be output to `info`, use `core.error` so that it's easy to find in the logs.